engine/engine.go

```diff
@@ -107,7 +107,8 @@ taskLoop:
 		default:
 		}
 
-		if e.pingWorker() {
+		check, _ := context.WithDeadline(context.Background(), time.Now().Add(3*time.Second))
```

Comment:
Since this context is only used within `apiGood()`, recommend creating it in there.  Also, do not ignore the cancel function so that cancel can be called as soon as ctx is not needed and its goroutine will be ended.